[LibWebRTC][Linux] VP9 encoding broken

philn opened the pull request:
#### 49b71e2d05c099f27f5b75a61bac0a1ea39daade
<pre>
[LibWebRTC][Linux] VP9 encoding broken
<a href="https://bugs.webkit.org/show_bug.cgi?id=302255">https://bugs.webkit.org/show_bug.cgi?id=302255</a>

Reviewed by Adrian Perez de Castro.

Enable the bundled libvpx build in CMake, thus ensuring the runtime ABI version will always be
compatible. The previous approach of relying on external libvpx was prone to ABI version mismatches
at runtime, thus disabling the VP9 encoder.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/303128@main">https://commits.webkit.org/303128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5790d51c52fcb42a1e5021e24506b2bb4150ea4e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131412 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3742 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42376 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138898 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83177 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3764 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3584 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/100339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134358 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/2715 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117629 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81127 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2631 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82110 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111241 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35752 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141540 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3487 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36272 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/108704 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; layout-tests (exception)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3533 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3116 "Found 1 new test failure: imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_pre.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108920 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27600 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2643 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113959 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56664 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3549 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32378 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3371 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3571 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3479 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/ThirdParty/libwebrtc/CMakeLists.txt:
```diff
@@ -4,16 +4,12 @@ set(libwebrtc_DERIVED_SOURCES_DIR "${CMAKE_BINARY_DIR}/libwebrtc/DerivedSources"
 add_definitions(-DLIBYUV_DISABLE_SVE)
 
 if (NOT APPLE)
-    find_package(LibVpx 1.10.0)
```
Actually I wonder how could this even work... we do not have a `Source/cmake/FindLibVpx.cmake` find-module, CMake does not have a built-in one, and distro packages do not ship a CMake find-module either 🤷🏼

aperezdc replied:
Ah, found it: `Source/ThirdParty/libwebrtc/cmake/FindLibVpx.cmake`

aperezdc commented on Source/ThirdParty/libwebrtc/CMakeLists.txt:
```diff
@@ -4,16 +4,12 @@ set(libwebrtc_DERIVED_SOURCES_DIR "${CMAKE_BINARY_DIR}/libwebrtc/DerivedSources"
 file(MAKE_DIRECTORY ${libwebrtc_DERIVED_SOURCES_DIR})
 
 enable_language(ASM)
+enable_language(ASM_NASM)
```
Wouldn't this be x86-only? My understanding is that this will try to find Nasm, and fail if not found. And it is definitely not needed for other architectures. If we want to check whether Nasm is usable, we should use [check_language()](https://cmake.org/cmake/help/latest/module/CheckLanguage.html#command:check_language) before attempting to enable it.

philn replied:
Right, I moved this to the x86 path!

aperezdc commented on Source/ThirdParty/libwebrtc/CMakeLists.txt:
```diff
@@ -3240,3 +2746,484 @@ if (APPLE)
+    )
+endif ()
+
+set_source_files_properties(${vpx_ASSEMBLY_SOURCES} PROPERTIES LANGUAGE ASM_NASM)
```
Shouldn't this be inside the `if (WTF_CPU_X86_64)` block above?

philn replied:
Indeed!

philn committed:
[LibWebRTC][Linux] VP9 encoding broken
https://bugs.webkit.org/show_bug.cgi?id=302255

Reviewed by Adrian Perez de Castro.

Enable the bundled libvpx build in CMake, thus ensuring the runtime ABI version will always be
compatible. The previous approach of relying on external libvpx was prone to ABI version mismatches
at runtime, thus disabling the VP9 encoder.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:

Canonical link: https://commits.webkit.org/303128@main
